Analysis/DEEP2_R23_O32/Plotting/more_plots.py

```diff
@@ -227,67 +176,79 @@ def dust_att_plot(combine_flux):
-    plt.show()
-    #fig.set_size_inches(8,8)
-    #fig.savefig(pdf_pages, format='pdf')
+    fig.set_size_inches(8, 8)
+    fig.savefig(pdf_pages, format='pdf')
 
-    fig,ax = plt.subplots()
-    ax.scatter(Gambet, R23, marker= '.')
+    fig, ax = plt.subplots()
+    ax.scatter(Gambet, R23, marker='.')
     ax.set_xlabel('H_gamma/H_beta')
     ax.set_ylabel('R23')
     ax.set_title('H_gamma/H_beta vs. R23')
-    plt.show()
-    #fig.set_size_inches(8,8)
-    #fig.savefig(pdf_pages, format='pdf')
-
+    fig.set_size_inches(8, 8)
+    fig.savefig(pdf_pages, format='pdf')
+    pdf_pages.close()
 
+    
+def plotting_individual_for_stacking_image(stack_spectra=False):
+    if not stack_spectra:
+        name = '/Users/reagenleimbach/Desktop/Zcalbase_gal/individual_plots_for_stacking_image.pdf'
+        RestframeMaster = '/Users/reagenleimbach/Desktop/Zcalbase_gal/Master_Grid.fits'
+        spec_range = range(100, 120)
+        y_lim = (-2, 2)
+        y_text = 1.95
+        left = 0.13
+    else:
+        RestframeMaster = '/Users/reagenleimbach/Desktop/Zcalbase_gal/R23O32_Manual_0417/Stacking_Masked_MasterGrid_n_Bins.fits'
+        name = '/Users/reagenleimbach/Desktop/Zcalbase_gal/R23O32_Manual_0417/composite_plots_data_viz.pdf'
+        spec_range = range(27)
+        y_lim = (0, 1)
+        y_text = 0.95
+        left = 0.1
 
-    #pdf_pages.close()
+    image2DM, header = fits.getdata(RestframeMaster, header=True)
+    wave = header['CRVAL1'] + header['CDELT1']*np.arange(header['NAXIS1'])
 
-    
-def plotting_individual_for_stacking_image():
-    name = '/Users/reagenleimbach/Desktop/Zcalbase_gal/individual_plots_for_stacking_image.pdf'
     pdf_pages = PdfPages(name)
-    
-    RestframeMaster = r'/Users/reagenleimbach/Desktop/Zcalbase_gal/Master_Grid.fits'
-    image2DM, header = fits.getdata(RestframeMaster, header=True)
-    wave= header['CRVAL1'] + header['CDELT1']*np.arange(header['NAXIS1'])
+
+    txt0 = r'Intensity ($10^{-17}~{\rm erg}~{\rm s}^{-1}~{\rm cm}^{-2}~\AA^{-1}$)'
 
     scalefactor = 1e-17
     image2d = image2DM/scalefactor
-    for ii in range(100,120):
-        fig,ax = plt.subplots()
-        plt.plot(wave, image2d[ii,:], linewidth = 0.5)
-        plt.xlabel('Wavelength')
-        plt.ylabel('Intensity Scale factor = 1e-17')
-        ax.set_xlim(4100,4500)
-                 
+    for ii in spec_range:
+        fig, ax = plt.subplots()
+        plt.plot(wave, image2d[ii, :], linewidth=0.5)
+        plt.xlabel('Wavelength (Angstroms)')
+        plt.ylabel(txt0)  # ergs per second per cm^2 per Angstrom
+        ax.set_xlim(4250, 4450)
+        ax.set_ylim(y_lim)
+        ax.axvline(x=4363.21, linewidth=1.0, color='r', linestyle=':')
+        ax.axvline(x=4340.544, linewidth=1.0, color='r', linestyle=':')
+        ax.text(4363.21, y_text, r'[OIII]$\lambda$4363', va='top', ha='center', rotation=90)
+        ax.text(4340.544, y_text, r'H$\gamma$', va='top', ha='center', rotation=90)
```

Comment:
Actually it should be 4340.46

Values are in MSC, so you can always import from there.